manifests/init.pp

```diff
@@ -21,27 +21,48 @@
     mode   => '0755',
   }
 
-  if $manage_database_maintenance {
-    include pe_databases::maintenance
+  exec { 'pe_databases_daemon_reload':
+    command     => 'systemctl daemon-reload',
+    path        => ['/bin', '/usr/bin'],
+    refreshonly => true,
   }
 
-  # Do not manage postgresql_settings in 2018.1.0 or newer.
-  if $manage_postgresql_settings and (versioncmp('2018.1.0', $facts['pe_server_version']) > 0) {
-    include pe_databases::postgresql_settings
-    class { 'pe_databases::postgresql_settings::table_settings' :
-      manage_reports_autovacuum_cost_delay    => $pe_databases::postgresql_settings::manage_reports_autovacuum_cost_delay,
-      factsets_autovacuum_vacuum_scale_factor => $pe_databases::postgresql_settings::factsets_autovacuum_vacuum_scale_factor,
-      reports_autovacuum_vacuum_scale_factor  => $pe_databases::postgresql_settings::reports_autovacuum_vacuum_scale_factor,
-      require                                 => Class['pe_databases::postgresql_settings'],
+  if $facts.dig('pe_databases', 'have_systemd') {
+    if $manage_database_maintenance and (versioncmp('2019.0.2', $facts['pe_server_version']) <= 0) {
+      class {'pe_databases::pg_repack':
+        disable_maintenance => lookup('pe_databases::maintenance::disable_maintenance', {'default_value' => false}),
+      }
+      # Do not manage postgresql_settings in 2018.1.0 or newer.
+      if $manage_postgresql_settings and (versioncmp('2018.1.0', $facts['pe_server_version']) > 0) {
```

Comment:
This is still checking for 2018, but never will be true since the check above is looking for 2019.0.2+

Reply:
Are you ok with removing the `versioncmp` here?

Reply:
We are dropping support for 2019 and below right? Let's drop the true condition here and only do the elseif, else part.

Reply:
:+1: